```
cl /Fowin64_build\plugins\win64_build\mediafoundation\soundsourcemediafoundation.obj /c plugins\soundsourcemediafoundation\soundsourcemediafoundation.cpp /TP /nologo /MP /Zi /MD /GL /fp:fast /Gy /O2 /Damd64 /DMIXXX_BUILD_RELEASE /D'Q_ASSERT(x)=qt_noop()' /D__WINDOWS__ /D_ATL_XP_TARGETING /D_ATL_MIN_CRT /DNOMINMAX /DUNICODE /DWIN64 /D_USE_MATH_DEFINES /DSETTINGS_FILE=\"mixxx.cfg\" /D__PORTAUDIO__ /DQT_TABLET_SUPPORT /DQT_SHARED /DQT_CORE_LIB /DQT_GUI_LIB /DQT_OPENGL_LIB /DQT_XML_LIB /DQT_SVG_LIB /DQT_SQL_LIB /DQT_SCRIPT_LIB /DQT_NETWORK_LIB /D__SNDFILE__ /DPROTOBUF_USE_DLLS /D__MAD__ /D__MEDIAFOUNDATION__ /D__HSS1394__ /D__HID__ /D__VINYLCONTROL__ /D__BROADCAST__ /D__OPUS__ /D__SSE__ /D__SSE2__ /D__SSE__ /D__SSE2__ /D__VAMP__ /Dkiss_fft_scalar=double /D__SQLITE3__ /D__BATTERY__ /D_USING_V110_SDK71_ /IC:\mixxx-buildserver\2.0-x64-release-minimal /IC:\mixxx-buildserver\2.0-x64-release-minimal\include /Iwin64_build\plugins\win64_build\mediafoundation /Iplugins\soundsourcemediafoundation /Isrc /Iplugins\win64_build\mediafoundation /Isrc\plugins\win64_build\mediafoundation /Ilib\soundtouch-1.9.2 /Ilib\replaygain /Ilib\libebur128-1.1.0\ebur128 /Ilib\libebur128-1.1.0\ebur128\queue /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtCore /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtGui /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtOpenGL /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtXml /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtSvg /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtSql /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtScript /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtXmlPatterns /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtNetwork /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtTest /IC:\mixxx-buildserver\2.0-x64-release-minimal\build\qt-everywhere-opensource-src-4.8.6\include\QtScriptTools /Ilib\gtest-1.7.0\include /Ilib\fidlib-0.9.10 /IC:\usr\include\taglib /Ilib\qtscript-bytearray /Ilib\reverb /Ilib\portaudio /Ilib\hidapi-0.8.0-rc1\hidapi /Ilib\xwax /Ilib\scratchlib /Ilib\vamp-2.6
cl @c:\users\appveyor\appdata\local\temp\1\tmpmtlctn.lnk
soundsourcemediafoundation.cpp
C:\projects\mixxx\src\util/audiosignal.h(76) : error C2610: 'mixxx::AudioSignal::AudioSignal(mixxx::AudioSignal &&)' : is not a special member function which can be defaulted
C:\projects\mixxx\src\util/audiosignal.h(78) : error C2610: 'mixxx::AudioSignal &mixxx::AudioSignal::operator =(mixxx::AudioSignal &&)' : is not a special member function which can be defaulted
```

--- a/src/library/coverart.h
+++ b/src/library/coverart.h
@@ -33,10 +33,13 @@ class CoverInfoRelative {
 
     // all-default memory management
     CoverInfoRelative(const CoverInfoRelative&) = default;
-    CoverInfoRelative(CoverInfoRelative&&) = default;
     CoverInfoRelative& operator=(const CoverInfoRelative&) = default;
-    CoverInfoRelative& operator=(CoverInfoRelative&&) = default;
     virtual ~CoverInfoRelative() = default;
+// Visual Studio does not support default generated move constructors
+#if !defined(_MSC_VER)
+    CoverInfoRelative(CoverInfoRelative&&) = default;
+    CoverInfoRelative& operator=(CoverInfoRelative&&) = default;
+#endif
 
     Source source;
     Type type;
@@ -59,10 +62,13 @@ class CoverInfo : public CoverInfoRelative {
 
     // all-default memory management
     CoverInfo(const CoverInfo&) = default;
-    CoverInfo(CoverInfo&&) = default;
     CoverInfo& operator=(const CoverInfo&) = default;
-    CoverInfo& operator=(CoverInfo&&) = default;
     virtual ~CoverInfo() override = default;
+// Visual Studio does not support default generated move constructors
+#if !defined(_MSC_VER)
+    CoverInfo(CoverInfo&&) = default;
+    CoverInfo& operator=(CoverInfo&&) = default;
+#endif
 
     QString trackLocation;
 };
@@ -85,10 +91,13 @@ class CoverArt : public CoverInfo {
 
     // all-default memory management
     CoverArt(const CoverArt&) = default;
-    CoverArt(CoverArt&&) = default;
     CoverArt& operator=(const CoverArt&) = default;
-    CoverArt& operator=(CoverArt&&) = default;
     virtual ~CoverArt() override = default;
+// Visual Studio does not support default generated move constructors
+#if !defined(_MSC_VER)
+    CoverArt(CoverArt&&) = default;
+    CoverArt& operator=(CoverArt&&) = default;
+#endif
 
     // it is not a QPixmap, because it is not safe to use pixmaps 
     // outside the GUI thread

--- a/src/util/audiosignal.h
+++ b/src/util/audiosignal.h
@@ -73,10 +73,13 @@ public:
 
     // all-default memory management
     AudioSignal(const AudioSignal&) = default;
-    AudioSignal(AudioSignal&&) = default;
     AudioSignal& operator=(const AudioSignal&) = default;
-    AudioSignal& operator=(AudioSignal&&) = default;
     virtual ~AudioSignal() = default;
+// Visual Studio does not support default generated move constructors
+#if !defined(_MSC_VER)
+    AudioSignal(AudioSignal&&) = default;
+    AudioSignal& operator=(AudioSignal&&) = default;
+#endif
 
     // Returns the ordering of samples in contiguous buffers
     SampleLayout getSampleLayout() const {

